lib/bas/bot/format_do_bill_alert.rb

```diff
@@ -48,17 +48,17 @@ class FormatDoBillAlert < Bot::Base
     # read function to execute the PostgresDB Read component
     #
     def read
-      reader = Read::Postgres.new(read_options.merge(conditions))
+      reader = Read::Postgres.new(read_options.merge(combined_conditions))
 
-      reader.execute
+      @previous_billing_data, @current_billing_data = split_billing_data(reader.execute)
     end
 
     # Process function to format the notification using a template
     #
     def process
-      return { success: { notification: "" } } if unprocessable_response || !threshold_exceeded
+      return { success: { notification: message } } if balance_alert?
 
-      { success: { notification: message } }
+      { success: { notification: "" } } unless unprocessable_response
```

Comment:
I think the condition should be `if unprocessable_response` since we would like to add an empty notification if the `read` step didn't find something to process. What do you think?